website_form_builder/migrations/1.2.0/post-migrate.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2020 Tecnativa - Jairo Llopis
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+
+from openupgradelib import openupgrade as ou
+from openupgradelib.openupgrade_tools import (
+    convert_html_fragment,
+    convert_html_replacement_class_shortcut as _r,
+)
+
+REPLACEMENTS = (
+    _r(
+        # Use XPath mode because LXML's CSS mode doesn't support `:has()`
+        selector="""
+        //*
+        [hasclass('form-field', 'o_required')]
+        [//input[not(@required)]]
```

Comment:
What this xpath does? Im lost with the [//input[not(@required)]] part...

All elements with 'form-field' & 'o_required' that have an input without required?

Reply:
> All elements with 'form-field' & 'o_required' that have an input without required?

exactly.